src/inmanta/server/services/compilerservice.py

```diff
@@ -336,20 +350,27 @@ async def install_modules() -> None:
             LOGGER.exception("An error occured while recompiling")
 
         finally:
+
+            async def warn(message: str) -> None:
+                if self.stage is not None:
```

Comment:
is there a code path that arrives here where there is no stage?

Reply:
To be honest I'm not completely sure. The thing is, there could be because we might jump from any place in the body above. In a case where it requires carefully reading a whole block of code to even determine if an `Optional` is ever `None` at the point of failure I prefer to reason on the type rather than the fragile knowledge of the actual value at the time of writing, with future changes to the try body in mind.
Here as well, I could take another approach if you prefer.